### ISSUE TITLE
Runtime error of inferred type with alias in its path

### ISSUE THREAD

**Issue by erdavila** [zio/zio#3390]
The code below compiles successfully, but fails with a runtime error:
```scala
import zio._

object Main extends App {
  type A = Has[A.Service]
  object A {
    trait Service
  }
  def access(n: Int): URIO[A, Int] = ZIO.access[A] { a => a.get; n }

  val B = A

  override def run(args: List[String]): ZIO[ZEnv, Nothing, Int] =
    access(0)
      .provideLayer(ZLayer.succeed(new B.Service {}))
}
```

The value `B` is an alias for `A`. So the type `B.Service` is the same as `A.Service`. However the code below fails with 
```
Fiber failed.
An unchecked error was produced.
java.lang.Error: Defect in zio.Has: Could not find Main::A::Service inside Map(B::Service -> Main$$anon$1@4ff12284)
```

It seems the `Tagged` instance for the inferred type parameter of `ZLayer.succeed` keeps the _path_ of the type (`B.Service` vs `A.Service`) instead of a canonical/normalized identifier for the type, as can be noticed in the error message.

**Comment by jdegoes** [comment]
This one looks like a bug in Izumi Reflect. /cc @neko-kai

**Comment by pshirshov** [comment]
Technically that's an expected behaviour. izumi-reflect support for PDTs is limited and probably there is no viable solution for this problem.

**Comment by jdegoes** [comment]
@pshirshov This is considered a path dependent type?

Is it possible to detect a PDT and error on such a thing, because type tags are not well-defined for PDTs?

**Comment by neko-kai** [comment]
@erdavila
This is a known limitation, paths in `izumi-reflect` are just variable names, as written. There is generally not enough information to equalize different paths, although in this case there is - you can try to add logic to choose the prefix of the singleton type of the prefix variable if it exists, somewhere around here: https://github.com/7mind/izumi/blob/develop/fundamentals/fundamentals-reflection/src/main/scala/izumi/fundamentals/reflection/macrortti/LightTypeTagImpl.scala#L479 if the usecase is very important for you

**Comment by pshirshov** [comment]
@jdegoes 

> This is considered a path dependent type?

Indeed it is, it has a prefix. Yup, we may detect&fail though not sure if it's a very good idea. Even limited support is better than no support at all. In this particular case error message is descriptive enough I think.

**Comment by pshirshov** [comment]
@neko-kai : I think there should be one more case in `unpackPrefix` falling back to the type for every variable.

**Comment by erdavila** [comment]
@pshirshov: I have not found a formal definition of PDTs, but when reading about path-dependent types I always see a type declared inside a **class** or a **trait** so that such type depends on an **instance** of the surrounding class or trait and the notation `ClassOrTrait#Type` is used.

The case here is the type `Service` declared inside the **object** `A`. There is no instance of `A` and, as it is in the code, the notation is `A.Service`.

Sorry if I'm being pick :grimacing: .

**Comment by pshirshov** [comment]
We fixed this particular problem in https://github.com/zio/izumi-reflect/commit/5c139acc533760e2459fbcdd57419855014834f4

**Comment by jdegoes** [comment]
@pshirshov Thank you, that's awesome! 🙏 ❤️

**Comment by neko-kai** [comment]
Fixed in izumi-reflect-0.13.0-M1
